Add content-length-range to pre-signed URL config

oliverroick opened the pull request:
This PR adds restrictions to the size of files that can be uploaded to S3 via pre-signed URLs. 

- When generating the pre-signed URL, `content-length-range` is added to the list of conditions allowing for file size between 0 bytes and a defined maximum. The allowed maximum can be defined in the settings by specifying `MAX_FILE_SIZE` in the AWS settings. If not specified it defaults to 1MB. 
- Adds changes to `script.js` to handle cases when the limit is exceeded gracefully.
- Adds changes to `test.views.fake_s3_upload` to mimic the behavior of S3 regarding oversize files in the dev environment.
- Adds the new setting to the docs in README

### Question

- Should we keep the default size at 1MB or should we choose something larger? Since we can adjust that value through settings, I don't think it's too important.

oliverroick committed:
Add content-length-range to presigned URL config

oliverroick committed:
Add file-size check to test.views.fake_s3_upload

oliverroick committed:
Add MAX_FILE_SIZE setting to README

alukach commented on buckets/storage.py:
```diff
@@ -18,6 +19,7 @@ def __init__(self):
         self.secret_key = settings.AWS['SECRET_KEY']
         self.region = settings.AWS['REGION']
         self.bucket_name = settings.AWS['BUCKET']
+        self.max_size = settings.AWS.get('MAX_FILE_SIZE', MAX_FILE_SIZE)
```
I would personally expect that there would be no default filesize to an S3 library.  I'd recommend defaulting to no filesize limit, only limiting if someone opts-in to this feature.

oliverroick replied:
That makes sense, I will change that.

amplifi replied:
Agreed 👍

alukach reviewed (approved):
Looks good. As mentioned, I think the library should default to no filesize restriction, however I'm not too opinionated on this.

amplifi commented on buckets/test/errors.py:
```diff
@@ -0,0 +1,2 @@
+EXCEED_MAX_SIZE = ("<Error><Code>EntityTooLarge</Code><Message>Your proposed "
```
I would display the MAX_FILE_SIZE value in the error message to the user.

amplifi commented on buckets/static/buckets/js/script.js:
```diff
@@ -103,9 +103,16 @@
+                
+                var xml = new DOMParser().parseFromString(response, "text/xml");
+                if (xml.getElementsByTagName('Code')[0].innerHTML === 'EntityTooLarge') {
+                    errorMsg += 'File size exceeds maximum limit.'
```
I would display the max file size in the error message to the user.

amplifi reviewed (approved):
Aside from adding the max value to the error message, LGTM 👍

oliverroick committed:
Remove default max size

oliverroick committed:
Show max size in error message

oliverroick commented:
@amplifi, @alukach —

- Removed the default limit.
- Added the max size to the error message. The size is displayed in MB rounded to two decimals. (I'm assuming people won't set the max size to 100 bytes).

amplifi commented:
Thanks, @oliverroick. Looks good! 🚢

oliverroick committed:
0.1.21
